server/src/main/java/org/elasticsearch/discovery/PeerFinder.java

```diff
@@ -248,33 +249,41 @@ private boolean handleWakeUp() {
             }
         });
 
-        transportService.getThreadPool().schedule(findPeersInterval, Names.GENERIC, new AbstractRunnable() {
-            @Override
-            public boolean isForceExecution() {
-                return true;
-            }
+        try {
+            transportService.getThreadPool().schedule(findPeersInterval, Names.GENERIC, new AbstractRunnable() {
+                @Override
+                public boolean isForceExecution() {
+                    return true;
+                }
 
-            @Override
-            public void onFailure(Exception e) {
-                assert false : e;
-                logger.debug("unexpected exception in wakeup", e);
-            }
+                @Override
+                public void onFailure(Exception e) {
+                    assert false : e;
+                    logger.debug("unexpected exception in wakeup", e);
+                }
 
-            @Override
-            protected void doRun() {
-                synchronized (mutex) {
-                    if (handleWakeUp() == false) {
-                        return;
+                @Override
+                protected void doRun() {
+                    synchronized (mutex) {
+                        if (handleWakeUp() == false) {
+                            return;
+                        }
                     }
+                    onFoundPeersUpdated();
                 }
-                onFoundPeersUpdated();
-            }
 
-            @Override
-            public String toString() {
-                return "PeerFinder handling wakeup";
+                @Override
+                public String toString() {
+                    return "PeerFinder handling wakeup";
+                }
+            });
+        } catch (EsRejectedExecutionException e) {
```

Comment:
The changes to this file are just for `EsRejectedExecutionException`-handling, which arose in test failures. No test to verify this specifically is added here.

Reply:
Can you explain what kind of test failures you were seeing? I would like to avoid this wrapping here.

Reply:
ok, got it now (after running into failure myself). As this pattern is repeated here many times, let's add a method to `ThreadPool` that schedules while ignoring shutdowns.

Reply:
Ok see 1b6278263a9162857d0c63dcac2fa6a173c754bb